packages/adapters/ethers/src/client.ts

```diff
@@ -717,13 +713,13 @@ export class EthersAdapter {
       }
     }
 
-    const chainChangedHandler = (networkId: string) => {
-      if (networkId) {
-        const networkIdNumber =
-          typeof networkId === 'string'
-            ? EthersHelpersUtil.hexStringToNumber(networkId)
-            : Number(networkId)
-        const caipNetwork = this.caipNetworks.find(c => c.chainId === networkIdNumber)
+    const chainChangedHandler = (chainId: string) => {
+      const chainIdNumber =
```

Comment:
We should control if the received `chainId` is already active, otherwise - still couldn't figure it out why - this listener is getting triggered multiple times in a single network switch and breaking the state

Same logic applied to all clients